plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -231,6 +257,77 @@ export async function createRouter(
 
   // NOTE: The v2 API is unstable
   router
+    .get(
+      '/v2/templates/:namespace/:kind/:name/parameter-schema',
+      async (req, res) => {
+        const { namespace, kind, name } = req.params;
+
+        if (namespace !== 'default') {
+          throw new InputError(
+            `Invalid namespace, only 'default' namespace is supported`,
+          );
+        }
+        if (kind.toLowerCase() !== 'template') {
+          throw new InputError(
+            `Invalid kind, only 'template' kind is supported`,
+          );
+        }
+
+        const template = await entityClient.findTemplate(name);
+        if (isBeta2Template(template)) {
+          const parameters = [template.spec.parameters ?? []].flat();
+          res.json({
+            title: template.metadata.title ?? template.metadata.name,
+            steps: parameters.map(schema => ({
+              title: schema.title ?? 'Fill in template parameters',
+              schema,
+            })),
+          });
+        } else if (isAlpha1Template(template)) {
+          res.json({
+            title: template.metadata.title ?? template.metadata.name,
+            steps: [
+              {
+                title: 'Fill in template parameters',
+                schema: template.spec.schema,
+              },
+              {
+                title: 'Choose owner and repo',
+                schema: {
+                  type: 'object',
+                  required: ['storePath', 'owner'],
+                  properties: {
+                    owner: {
+                      type: 'string',
+                      title: 'Owner',
+                      description: 'Who is going to own this component',
```

Comment:
codeowners and stuff ;) but yeah for legacy it doesn't matter

Reply:
Copied this from the frontend so let's keep it the same